ext/strings_test.go

```diff
@@ -329,3 +330,266 @@ func TestStringsWithExtension(t *testing.T) {
 		t.Fatalf("env.Extend(Strings()) failed: %v", err)
 	}
 }
+
+func TestStringFormat(t *testing.T) {
+	tests := []struct {
+		name   string
+		format string
+		//formatArgs     []ref.Val
+		formatArgs     string
+		locale         string
+		err            string
+		expectedOutput string
+	}{
+		{
+			name:           "no-op",
+			format:         "no substitution",
+			expectedOutput: "no substitution",
+		},
+
+		{
+			name:           "mid-string substitution",
+			format:         "str is %s and some more",
+			formatArgs:     `"filler"`,
+			expectedOutput: "str is filler and some more",
+		},
+		{
+			name:           "percent escaping",
+			format:         "%% and also %%",
+			expectedOutput: "% and also %",
+		},
+		{
+			name:           "substution inside escaped percent signs",
+			format:         "%%%s%%",
+			formatArgs:     `"text"`,
+			expectedOutput: "%text%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the right",
+			format:         "%s%%",
+			formatArgs:     `"percent on the right"`,
+			expectedOutput: "percent on the right%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the left",
+			format:         "%%%s",
+			formatArgs:     `"percent on the left"`,
+			expectedOutput: "%percent on the left",
+		},
+		{
+			name:           "multiple substitutions",
+			format:         "%d %d %d, %s %s %s, %d %d %d, %s %s %s",
```

Comment:
Test `uint(<some integer)` somewhere?

Reply:
Right, I'll add a uint test and mark this [and this](https://github.com/google/cel-go/pull/617#discussion_r1097646454) as resolved once I do so.